### PR TITLE
feat(automation): github actions deploy foundation

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# Description

The scope of the changes in this pull request are to implement the deploy workflow foundation needed to override the existing deploy workflows with proper version control available via GHA instead of CircleCI.

## Links

[SPARK-429742](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-429742)

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+          key: node-modules-${{ hashFiles('./yarn.lock') }}
+
+      - name: Synchronize Packages
+        run: yarn
+
+      - name: Build Tools
+        run: yarn workspaces foreach --from '@webex/*-tools' --topological-dev --parallel --verbose run build:src
+
+      - name: Build # Optional, will remove this step once validated.
+        run: yarn workspaces foreach --parallel --verbose run build:src
```
Comment (arun3528) [inline .github/workflows/deploy.yml:68]: a deploy can run at a later state , shouldn't we do build again later for individual package

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+          path: '**/dist'
+          key: dist-${{ env.rid }}
+
+      - name: Synchronize Packages
+        run: yarn
+
+      - name: Update Main Branch
+        run: git fetch origin ${{ github.event.base_ref }} --unshallow
+
+      - id: generate-package-matrix-node-changed
```
Comment (arun3528) [inline .github/workflows/deploy.yml:117]: can we add comments to the package Metrix logic

**Review by arun3528 — COMMENTED** [review]
The code looks good just needs some description for the team on the package matrix part

**Comment by InteractiveTimmy** [comment]
@arun3528 , I've added some descriptions and updated some other comments, please let me know if there's anything else I should change.

**Review by arun3528 — APPROVED** [review]

**Pull request merged**